From 97a00fa6fc73600e02d60ea195d0990a3b99255a Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 11:51:23 -0500
Subject: [PATCH 01/83] round 1

---
 .scrutinizer.yml | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index 7517a05..e47c79f 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -35,3 +35,26 @@ checks:
         one_class_per_file: false
         avoid_superglobals: false
         require_php_tag_first: false
+        code_rating: true
+        duplication: true
+        variable_existence: true
+        useless_calls: true
+        use_statement_alias_conflict: true
+        unused_variables: true
+        unused_properties: true
+        unused_parameters: true
+        unused_methods: true
+        unreachable_code: true
+        sql_injection_vulnerabilities: true
+        security_vulnerabilities: true
+        precedence_mistakes: true
+        precedence_in_conditions: true
+        parameter_non_unique: true
+        no_property_on_interface: true
+        no_non_implemented_abstract_methods: true
+        deprecated_code_usage: true
+        closure_use_not_conflicting: true
+        closure_use_modifiable: true
+        avoid_useless_overridden_methods: true
+        avoid_conflicting_incrementers: true
+        assignment_of_null_return: true

From 9fe1f2f58f3325a1d60b48597b315eb0f66e5165 Mon Sep 17 00:00:00 2001
From: Scrutinizer Auto-Fixer <auto-fixer@scrutinizer-ci.com>
Date: Sat, 14 Feb 2015 17:22:09 +0000
Subject: [PATCH 02/83] Scrutinizer Auto-Fixes

This commit consists of patches automatically generated for this project on https://scrutinizer-ci.com
---
 includes/admin/downloads/metabox.php                           |  2 +-
 includes/admin/payments/actions.php                            |  1 -
 .../admin/reporting/class-api-requests-logs-list-table.php     |  2 +-
 includes/admin/reporting/class-sales-logs-list-table.php       |  2 +-
 includes/class-edd-api.php                                     |  2 +-
 includes/class-edd-download.php                                |  2 ++
 includes/class-edd-session.php                                 | 10 +++++-----
 includes/discount-functions.php                                |  4 ++--
 includes/emails/class-edd-emails.php                           |  1 +
 includes/misc-functions.php                                    |  2 +-
 includes/template-functions.php                                |  2 +-
 11 files changed, 16 insertions(+), 14 deletions(-)

diff --git a/includes/admin/downloads/metabox.php b/includes/admin/downloads/metabox.php
index fe4d270..0e43ad9 100755
--- a/includes/admin/downloads/metabox.php
+++ b/includes/admin/downloads/metabox.php
@@ -167,7 +167,7 @@ function edd_download_meta_box_save( $post_id, $post ) {
  *
  * @since 1.3.2
  * @param string $price Price before sanitization
- * @return float $price Sanitized price
+ * @return string $price Sanitized price
  */
 function edd_sanitize_price_save( $price ) {
 	return edd_sanitize_amount( $price );
diff --git a/includes/admin/payments/actions.php b/includes/admin/payments/actions.php
index 7e4c0c0..53b8b54 100644
--- a/includes/admin/payments/actions.php
+++ b/includes/admin/payments/actions.php
@@ -311,7 +311,6 @@ function edd_trigger_payment_note_deletion( $data ) {
  * Delete a payment note deletion with ajax
  *
  * @since 1.6
- * @param array $data Arguments passed
  * @return void
 */
 function edd_ajax_delete_payment_note() {
diff --git a/includes/admin/reporting/class-api-requests-logs-list-table.php b/includes/admin/reporting/class-api-requests-logs-list-table.php
index 518b790..2ba228d 100755
--- a/includes/admin/reporting/class-api-requests-logs-list-table.php
+++ b/includes/admin/reporting/class-api-requests-logs-list-table.php
@@ -151,7 +151,7 @@ public function column_details( $item ) {
 	 *
 	 * @access public
 	 * @since 1.5
-	 * @return mixed String if search is present, false otherwise
+	 * @return string|false String if search is present, false otherwise
 	 */
 	public function get_search() {
 		return ! empty( $_GET['s'] ) ? urldecode( trim( $_GET['s'] ) ) : false;
diff --git a/includes/admin/reporting/class-sales-logs-list-table.php b/includes/admin/reporting/class-sales-logs-list-table.php
index 1ad53d8..bfe76dc 100755
--- a/includes/admin/reporting/class-sales-logs-list-table.php
+++ b/includes/admin/reporting/class-sales-logs-list-table.php
@@ -141,7 +141,7 @@ public function get_filtered_download() {
 	 *
 	 * @access public
 	 * @since 1.4
-	 * @return mixed string If search is present, false otherwise
+	 * @return string|false string If search is present, false otherwise
 	 */
 	public function get_search() {
 		return ! empty( $_GET['s'] ) ? urldecode( trim( $_GET['s'] ) ) : false;
diff --git a/includes/class-edd-api.php b/includes/class-edd-api.php
index 190747a..0c5a787 100755
--- a/includes/class-edd-api.php
+++ b/includes/class-edd-api.php
@@ -140,7 +140,7 @@ public function add_endpoint( $rewrite_rules ) {
 	 * @since 1.5
 	 * @author Daniel J Griffiths
 	 * @param array $vars Query vars
-	 * @return array $vars New query vars
+	 * @return string[] $vars New query vars
 	 */
 	public function query_vars( $vars ) {
 		$vars[] = 'token';
diff --git a/includes/class-edd-download.php b/includes/class-edd-download.php
index 391828e..e0418d4 100644
--- a/includes/class-edd-download.php
+++ b/includes/class-edd-download.php
@@ -254,6 +254,7 @@ public function has_variable_prices() {
 	 * Retrieve the file downloads
 	 *
 	 * @since 2.2
+	 * @param integer $variable_price_id
 	 * @return array
 	 */
 	public function get_files( $variable_price_id = null ) {
@@ -590,6 +591,7 @@ public function increase_earnings( $amount = 0 ) {
 	 * Decrease the earnings by the given amount
 	 *
 	 * @since 2.2
+	 * @param integer $amount
 	 * @return float|false
 	 */
 	public function decrease_earnings( $amount ) {
diff --git a/includes/class-edd-session.php b/includes/class-edd-session.php
index 6e62194..f0d56dc 100755
--- a/includes/class-edd-session.php
+++ b/includes/class-edd-session.php
@@ -161,9 +161,9 @@ public function get( $key ) {
 	 *
 	 * @since 1.5
 	 *
-	 * @param $key Session key
-	 * @param $value Session variable
-	 * @return mixed Session variable
+	 * @param string $key Session key
+	 * @param integer $value Session variable
+	 * @return string Session variable
 	 */
 	public function set( $key, $value ) {
 
@@ -230,14 +230,14 @@ public function set_expiration_time( $exp ) {
 	/**
 	 * Starts a new session if one hasn't started yet.
 	 *
-	 * @return null
+	 * @return boolean
 	 * Checks to see if the server supports PHP sessions
 	 * or if the EDD_USE_PHP_SESSIONS constant is defined
 	 *
 	 * @access public
 	 * @since 2.1
 	 * @author Daniel J Griffiths
-	 * @return bool $ret True if we are using PHP sessions, false otherwise
+	 * @return boolean $ret True if we are using PHP sessions, false otherwise
 	 */
 	public function use_php_sessions() {
 
diff --git a/includes/discount-functions.php b/includes/discount-functions.php
index 44a98b7..220aec0 100755
--- a/includes/discount-functions.php
+++ b/includes/discount-functions.php
@@ -85,7 +85,7 @@ function edd_has_active_discounts() {
  * Retrieves a complete discount code by discount ID.
  *
  * @since 1.0
- * @param string $discount_id Discount ID
+ * @param integer $discount_id Discount ID
  * @return array
  */
 function edd_get_discount( $discount_id = 0 ) {
@@ -946,7 +946,7 @@ function edd_format_discount_rate( $type, $amount ) {
  *
  * @since 1.4.1
  * @param string $code Discount code
- * @return array All currently active discounts
+ * @return string[] All currently active discounts
  */
 function edd_set_cart_discount( $code = '' ) {
 
diff --git a/includes/emails/class-edd-emails.php b/includes/emails/class-edd-emails.php
index 3330132..5f693dc 100644
--- a/includes/emails/class-edd-emails.php
+++ b/includes/emails/class-edd-emails.php
@@ -191,6 +191,7 @@ public function get_heading() {
 	 * Parse email template tags
 	 *
 	 * @since 2.1
+	 * @param string $content
 	 */
 	public function parse_tags( $content ) {
 
diff --git a/includes/misc-functions.php b/includes/misc-functions.php
index 40533b5..6bcede0 100755
--- a/includes/misc-functions.php
+++ b/includes/misc-functions.php
@@ -403,7 +403,7 @@ function edd_get_currency_name( $code = 'USD' ) {
  *
  * @since 1.0
  *
- * @param unknown $n
+ * @param integer $n
  * @return string Short month name
  */
 function edd_month_num_to_name( $n ) {
diff --git a/includes/template-functions.php b/includes/template-functions.php
index 7e3767c..8847f70 100755
--- a/includes/template-functions.php
+++ b/includes/template-functions.php
@@ -753,7 +753,7 @@ function edd_is_purchase_history_page() {
  * Adds body classes for EDD pages
  *
  * @since 2.1
- * @param array $classes current classes
+ * @param array $class current classes
  * @return array Modified array of classes
  */
 function edd_add_body_classes( $class ) {

From bb909efc8210d5c472ed6b0b4f0c3c8750d1b47c Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 12:23:19 -0500
Subject: [PATCH 03/83] Docbloc fixes

---
 includes/admin/payments/class-payments-table.php                  | 2 +-
 includes/admin/reporting/class-api-requests-logs-list-table.php   | 2 +-
 includes/admin/reporting/class-file-downloads-logs-list-table.php | 2 +-
 includes/admin/reporting/class-gateway-error-logs-list-table.php  | 2 +-
 includes/admin/reporting/class-sales-logs-list-table.php          | 2 +-
 includes/class-edd-api.php                                        | 7 ++++---
 6 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/includes/admin/payments/class-payments-table.php b/includes/admin/payments/class-payments-table.php
index 5230579..866d7f8 100755
--- a/includes/admin/payments/class-payments-table.php
+++ b/includes/admin/payments/class-payments-table.php
@@ -252,7 +252,7 @@ public function get_sortable_columns() {
 	 * @access public
 	 * @since 1.4
 	 *
-	 * @param array $item Contains all the data of the discount code
+	 * @param array $payment Contains all the data of the payment
 	 * @param string $column_name The name of the column
 	 *
 	 * @return string Column Name
diff --git a/includes/admin/reporting/class-api-requests-logs-list-table.php b/includes/admin/reporting/class-api-requests-logs-list-table.php
index 2ba228d..9fce0cc 100755
--- a/includes/admin/reporting/class-api-requests-logs-list-table.php
+++ b/includes/admin/reporting/class-api-requests-logs-list-table.php
@@ -101,7 +101,7 @@ public function get_columns() {
 	 * @access public
 	 * @since 1.5
 	 *
-	 * @param array $item Contains all the data of the discount code
+	 * @param array $item Contains all the data of the api request
 	 * @param string $column_name The name of the column
 	 *
 	 * @return string Column Name
diff --git a/includes/admin/reporting/class-file-downloads-logs-list-table.php b/includes/admin/reporting/class-file-downloads-logs-list-table.php
index cbd2b9c..dbc7866 100755
--- a/includes/admin/reporting/class-file-downloads-logs-list-table.php
+++ b/includes/admin/reporting/class-file-downloads-logs-list-table.php
@@ -102,7 +102,7 @@ public function search_box( $text, $input_id ) {
 	 * @access public
 	 * @since 1.4
 	 *
-	 * @param array $item Contains all the data of the discount code
+	 * @param array $item Contains all the data of the log item
 	 * @param string $column_name The name of the column
 	 *
 	 * @return string Column Name
diff --git a/includes/admin/reporting/class-gateway-error-logs-list-table.php b/includes/admin/reporting/class-gateway-error-logs-list-table.php
index ca9538d..fbde478 100755
--- a/includes/admin/reporting/class-gateway-error-logs-list-table.php
+++ b/includes/admin/reporting/class-gateway-error-logs-list-table.php
@@ -56,7 +56,7 @@ public function __construct() {
 	 * @access public
 	 * @since 1.4
 	 *
-	 * @param array $item Contains all the data of the discount code
+	 * @param array $item Contains all the data of the log item
 	 * @param string $column_name The name of the column
 	 *
 	 * @return string Column Name
diff --git a/includes/admin/reporting/class-sales-logs-list-table.php b/includes/admin/reporting/class-sales-logs-list-table.php
index bfe76dc..a38b5d0 100755
--- a/includes/admin/reporting/class-sales-logs-list-table.php
+++ b/includes/admin/reporting/class-sales-logs-list-table.php
@@ -57,7 +57,7 @@ public function __construct() {
 	 * @access public
 	 * @since 1.4
 	 *
-	 * @param array $item Contains all the data of the discount code
+	 * @param array $item Contains all the data of the log item
 	 * @param string $column_name The name of the column
 	 *
 	 * @return string Column Name
diff --git a/includes/class-edd-api.php b/includes/class-edd-api.php
index 0c5a787..0cbddeb 100755
--- a/includes/class-edd-api.php
+++ b/includes/class-edd-api.php
@@ -1459,8 +1459,9 @@ public function process_api_key( $args ) {
 	 *
 	 * @access public
 	 * @since 2.0.0
-	 * @param array $args
-	 * @return string
+	 * @param int $user_id User ID the key is being generated for
+	 * @param boolean $regenerate Regenerate the key for the user
+	 * @return boolean True if (re)generated succesfully, false otherwise.
 	 */
 	public function generate_api_key( $user_id = 0, $regenerate = false ) {
 
@@ -1493,7 +1494,7 @@ public function generate_api_key( $user_id = 0, $regenerate = false ) {
 	 *
 	 * @access public
 	 * @since 2.0.0
-	 * @param int $args
+	 * @param int $user_id User ID of user to revoke key for
 	 * @return string
 	 */
 	public function revoke_api_key( $user_id = 0 ) {

From 4fc25edb3297772a5c3074112f691509e4c24d85 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 12:23:30 -0500
Subject: [PATCH 04/83] More fixes

---
 includes/class-edd-html-elements.php | 13 +++----------
 includes/discount-functions.php      |  2 +-
 includes/misc-functions.php          |  2 +-
 includes/user-functions.php          |  2 +-
 5 files changed, 6 insertions(+), 13 deletions(-)

diff --git a/includes/class-edd-html-elements.php b/includes/class-edd-html-elements.php
index 751fe0e..b663bcd 100755
--- a/includes/class-edd-html-elements.php
+++ b/includes/class-edd-html-elements.php
@@ -26,8 +26,7 @@ class EDD_HTML_Elements {
 	 *
 	 * @access public
 	 * @since 1.5
-	 * @param string $name Name attribute of the dropdown
-	 * @param int $selected Download to select automatically
+	 * @param array $args Arguments for the dropdown
 	 * @return string $output Product dropdown
 	 */
 	public function product_dropdown( $args = array() ) {
@@ -396,10 +395,7 @@ public function checkbox( $args = array() ) {
 	 *
 	 * @since 1.5.2
 	 *
-	 * @param string $name Name attribute of the text field
-	 * @param string $value The value to prepopulate the field with
-	 * @param string $label
-	 * @param string $desc
+	 * @param array $args Arguments for the text field
 	 * @return string Text field
 	 */
 	public function text( $args = array() ) {
@@ -451,10 +447,7 @@ public function text( $args = array() ) {
 	 *
 	 * @since 1.9
 	 *
-	 * @param string $name Name attribute of the textarea
-	 * @param string $value The value to prepopulate the field with
-	 * @param string $label
-	 * @param string $desc
+	 * @param array $args Arguments for the textarea
 	 * @return string textarea
 	 */
 	public function textarea( $args = array() ) {
diff --git a/includes/discount-functions.php b/includes/discount-functions.php
index 220aec0..953b386 100755
--- a/includes/discount-functions.php
+++ b/includes/discount-functions.php
@@ -530,7 +530,7 @@ function edd_get_discount_product_condition( $code_id = 0 ) {
  * @since 1.5
  * @param int $code_id Discount ID
  * @return array $product_reqs IDs of the required products
- * @return bool Whether or not discount code is global
+ * @return boolean Whether or not discount code is global
  */
 function edd_is_discount_not_global( $code_id = 0 ) {
 	return (bool) get_post_meta( $code_id, '_edd_discount_is_not_global', true );
diff --git a/includes/misc-functions.php b/includes/misc-functions.php
index 6bcede0..dcf7699 100755
--- a/includes/misc-functions.php
+++ b/includes/misc-functions.php
@@ -387,7 +387,7 @@ function edd_currency_symbol( $currency = '' ) {
  * Get the name of a currency
  *
  * @since 2.2
- * @param  string $currency The currency code
+ * @param  string $code The currency code
  * @return string The currency's name
  */
 function edd_get_currency_name( $code = 'USD' ) {
diff --git a/includes/user-functions.php b/includes/user-functions.php
index 6aae38e..4202b63 100755
--- a/includes/user-functions.php
+++ b/includes/user-functions.php
@@ -334,7 +334,7 @@ function edd_count_file_downloads_of_user( $user ) {
  *
  * @access      public
  * @since       1.3.4
- * @param       $username string - the username to validate
+ * @param       string $username The username to validate
  * @return      bool
  */
 function edd_validate_username( $username ) {


From c7ba7a2c27c3b3922d6f8e1113e36e9eea95e81a Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 12:29:54 -0500
Subject: [PATCH 06/83] Syntax error

---
 includes/class-edd-cli.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/class-edd-cli.php b/includes/class-edd-cli.php
index 6527e07..9f38cee 100644
--- a/includes/class-edd-cli.php
+++ b/includes/class-edd-cli.php
@@ -63,7 +63,7 @@ public function details( $args, $assoc_args ) {
 		WP_CLI::line( sprintf( __( 'Ajax is: %s', 'edd' ), ( edd_is_ajax_enabled() ? __( 'Enabled', 'edd' ) : __( 'Disabled', 'edd' ) ) ) );
 		WP_CLI::line( sprintf( __( 'Guest checkouts are: %s', 'edd' ), ( edd_no_guest_checkout() ? __( 'Disabled', 'edd' ) : __( 'Enabled', 'edd' ) ) ) );
 		WP_CLI::line( sprintf( __( 'Symlinks are: %s', 'edd' ), ( apply_filters( 'edd_symlink_file_downloads', isset( $symlink_file_downloads ) ) && function_exists( 'symlink' ) ? __( 'Enabled', 'edd' ) : __( 'Disabled', 'edd' ) ) ) );
-		WP_CLI::line( "\n" . sprintf( __( 'Checkout page is: %s', 'edd' ), ( ! edd_get_option( 'purchase_page', false ) ) ? __( 'Valid', 'edd' ) : __( 'Invalid', 'edd' ) ) ) );
+		WP_CLI::line( "\n" . sprintf( __( 'Checkout page is: %s', 'edd' ), ( ! edd_get_option( 'purchase_page', false ) ) ? __( 'Valid', 'edd' ) : __( 'Invalid', 'edd' ) ) );
 		WP_CLI::line( sprintf( __( 'Checkout URL is: %s', 'edd' ), ( ! empty( $purchase_page ) ? get_permalink( $purchase_page ) : __( 'Undefined', 'edd' ) ) ) );
 		WP_CLI::line( sprintf( __( 'Success URL is: %s', 'edd' ), ( ! empty( $success_page ) ? get_permalink( $success_page ) : __( 'Undefined', 'edd' ) ) ) );
 		WP_CLI::line( sprintf( __( 'Failure URL is: %s', 'edd' ), ( ! empty( $failure_page ) ? get_permalink( $failure_page ) : __( 'Undefined', 'edd' ) ) ) );

From f7761956c35d74418831aea1bc026aff4de45c0e Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 12:33:03 -0500
Subject: [PATCH 07/83] Constructors don't return values

---
 includes/admin/tracking.php | 1 -
 1 file changed, 1 deletion(-)

diff --git a/includes/admin/tracking.php b/includes/admin/tracking.php
index c4396df..b7eb6e6 100755
--- a/includes/admin/tracking.php
+++ b/includes/admin/tracking.php
@@ -33,7 +33,6 @@ class EDD_Tracking {
 	 * Get things going
 	 *
 	 * @access public
-	 * @return void
 	 */
 	public function __construct() {
 
From 99ee705e1f705b12e38d587e7a1fc5c2e909edf0 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 12:35:34 -0500
Subject: [PATCH 10/83] Constructors don't return values

---
 includes/EDD_SL_Plugin_Updater.php | 1 -
 1 file changed, 1 deletion(-)

diff --git a/includes/EDD_SL_Plugin_Updater.php b/includes/EDD_SL_Plugin_Updater.php
index c000383..ab392b3 100644
--- a/includes/EDD_SL_Plugin_Updater.php
+++ b/includes/EDD_SL_Plugin_Updater.php
@@ -27,7 +27,6 @@ class EDD_SL_Plugin_Updater {
      * @param string  $_api_url     The URL pointing to the custom API endpoint.
      * @param string  $_plugin_file Path to the plugin file.
      * @param array   $_api_data    Optional data to send with API calls.
-     * @return void
      */
     function __construct( $_api_url, $_plugin_file, $_api_data = null ) {
         $this->api_url  = trailingslashit( $_api_url );

From 91d57dfc9899775f6df176e96ff8ae7f45a82719 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 12:41:57 -0500
Subject: [PATCH 11/83] Initialize array before use

---
 includes/user-functions.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/includes/user-functions.php b/includes/user-functions.php
index 4202b63..bc6f9d4 100755
--- a/includes/user-functions.php
+++ b/includes/user-functions.php
@@ -256,6 +256,7 @@ function edd_get_purchase_stats_by_user( $user = '' ) {
 	$customer = EDD()->customers->get_customer_by( $field, $user );
 	$customer = new EDD_Customer( $customer->id );
 
+	$stats = array();
 	$stats['purchases']   = absint( $customer->purchase_count );
 	$stats['total_spent'] = edd_sanitize_amount( $customer->purchase_value );

From d052bf313fcd91f1e1cfb1bfc6e3feeb93f2f6f6 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 12:57:52 -0500
Subject: [PATCH 13/83] Add irc notifications back. Exclude non compatible hhvm
 versions. Add fast finish

---
 .travis.yml | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/.travis.yml b/.travis.yml
index 2cf690f..4de6870 100755
--- a/.travis.yml
+++ b/.travis.yml
@@ -17,8 +17,14 @@ env:
 
 matrix:
   include:
-  - php: 5.3
+    - php: 5.3
     env: WP_VERSION=latest WP_MULTISITE=1
+  exclude:
+    - php: hhvm
+    env: WP_VERSION=3.9.2 WP_MULTISITE=1
+    - php: hhvm
+    env: WP_VERSION=4.0 WP_MULTISITE=1
+  fast_finish: true
 
 before_script:
 - bash bin/install-wp-tests.sh wordpress_test root '' localhost $WP_VERSION
@@ -37,3 +43,8 @@ after_script:
 notifications:
   slack:
     secure: g32q5XHgjZI7eNpNEBRpVab9f1JNoxBP6q5j+v/8qhbp9mcHGyfAjoPePCCB2IPtDByMAhRg/RmKwtj0yt1ORqftcbCFrMni6ePBSBRzjFIIsYlEZbEr2m1bRZTHpYgkVvkNCFra6y7kzL8WsnYPokiITRHw5/BKxLdePEgk1Q0=
+  irc:
+    channels:
+        - "irc.freenode.net#eddwp"
+    template:
+        - "Build %{build_number} (%{branch} - %{commit}): %{message} %{build_url}"
\ No newline at end of file

From f346375602627e2ef1197f25bc8a1700a7c03559 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:10:46 -0500
Subject: [PATCH 14/83] Remove old travis file

---
 .travis-old.yml | 52 ----------------------------------------------------
 1 file changed, 52 deletions(-)
 delete mode 100644 .travis-old.yml

diff --git a/.travis-old.yml b/.travis-old.yml
deleted file mode 100644
index c7dc6ab..0000000
--- a/.travis-old.yml
+++ /dev/null
@@ -1,52 +0,0 @@
-language: php
-
-php:
-    - 5.3
-    - 5.4
-
-env:
-    - WP_VERSION=trunk/ WP_MULTISITE=0
-    - WP_VERSION=tags/3.9/ WP_MULTISITE=0
-    - WP_VERSION=tags/3.8/ WP_MULTISITE=0
-    - WP_VERSION=trunk/ WP_MULTISITE=1
-    - WP_VERSION=tags/3.9/ WP_MULTISITE=1
-    - WP_VERSION=tags/3.8/ WP_MULTISITE=1
-
-matrix:
-    allow_failures:
-        - WP_VERSION=tags/3.8/ WP_MULTISITE=0
-        - WP_VERSION=tags/3.8/ WP_MULTISITE=1
-    fast_finish: true
-    
-
-before_script:
-    - mkdir -p tmp
-    - WP_CORE_DIR=tmp/wordpress/
-    - WP_TESTS_DIR=tmp/wordpress-tests/
-    - svn co --ignore-externals https://core.svn.wordpress.org/$WP_VERSION tmp/wordpress/
-    - svn co --ignore-externals https://unit-tests.svn.wordpress.org/trunk/ tmp/wordpress-tests/
-    - cp tmp/wordpress-tests/wp-tests-config-sample.php tmp/wordpress-tests/wp-tests-config.php
-    - cp -fr tests/bootstrap-wordpress-unit-test-suite.php tmp/wordpress-tests/includes/bootstrap.php
-    - sed -i "s:dirname( __FILE__ ) . '/wordpress/':'$WP_CORE_DIR':" tmp/wordpress-tests/wp-tests-config.php
-    - sed -i "s/yourdbnamehere/wordpress_test/" tmp/wordpress-tests/wp-tests-config.php
-    - sed -i "s/yourusernamehere/root/" tmp/wordpress-tests/wp-tests-config.php
-    - sed -i "s/yourpasswordhere//" tmp/wordpress-tests/wp-tests-config.php
-    - mysql -e 'CREATE DATABASE wordpress_test;' -uroot
-    - wget https://getcomposer.org/composer.phar
-    - php composer.phar install --dev --no-interaction
-    - phpenv rehash
-
-script:
-    - mkdir -p build/logs
-    - phpunit -c phpunit.xml
-
-after_script:
-    - php vendor/bin/coveralls --config tests/framework/.coveralls.yml -v --exclude-no-stmt
-
-notifications:
-    slack: chriscct7:yd20WubfRqEZM1vlsEhN7E7N
-    irc:
-        channels:
-            - "irc.freenode.net#eddwp"
-        template:
-            - "Build %{build_number} (%{branch} - %{commit}): %{message} %{build_url}"

From 88065bb95eda33003072b4589ae33a9501cbb065 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:10:57 -0500
Subject: [PATCH 15/83] Up the timeout on code coverage

---
 .scrutinizer.yml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index e47c79f..14f3e8e 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -3,7 +3,8 @@ filter:
 before_commands:
     - 'composer install --dev --prefer-source'
 tools:
-    external_code_coverage: true
+    external_code_coverage:
+            timeout: 900    # wait for 15 minutes instead of 5
     php_mess_detector: 
         config:
             code_size_rules:

From 5fbcac5c32e741b31bad99e28416ef6ac0ec8c27 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:29:45 -0500
Subject: [PATCH 16/83] remove newline

---
 .scrutinizer.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index 14f3e8e..7e4951f 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -58,4 +58,4 @@ checks:
         closure_use_modifiable: true
         avoid_useless_overridden_methods: true
         avoid_conflicting_incrementers: true
-        assignment_of_null_return: true
+        assignment_of_null_return: true
\ No newline at end of file

From 5d96f7c0c99249528d8660aac3a2198ab7ec6d53 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:39:14 -0500
Subject: [PATCH 17/83] Validated

---
 .travis.yml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/.travis.yml b/.travis.yml
index 4de6870..2e0cd00 100755
--- a/.travis.yml
+++ b/.travis.yml
@@ -18,12 +18,12 @@ env:
 matrix:
   include:
     - php: 5.3
-    env: WP_VERSION=latest WP_MULTISITE=1
+      env: WP_VERSION=latest WP_MULTISITE=1
   exclude:
     - php: hhvm
-    env: WP_VERSION=3.9.2 WP_MULTISITE=1
+      env: WP_VERSION=3.9.2 WP_MULTISITE=1
     - php: hhvm
-    env: WP_VERSION=4.0 WP_MULTISITE=1
+      env: WP_VERSION=4.0 WP_MULTISITE=1
   fast_finish: true
 
 before_script:

From 15e80e710f9f0327684455006923131926597221 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:46:03 -0500
Subject: [PATCH 18/83] Fixes start_date bug in WP CLI

---
 includes/class-edd-cli.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/class-edd-cli.php b/includes/class-edd-cli.php
index 9f38cee..4a50764 100644
--- a/includes/class-edd-cli.php
+++ b/includes/class-edd-cli.php
@@ -108,7 +108,7 @@ public function stats( $args, $assoc_args ) {
 		if( ! empty( $date ) ) {
 			$start_date = $date;
 			$end_date   = false;
-		} elseif( empty( $date ) && empty( $startdate ) ) {
+		} elseif( empty( $date ) && empty( $start_date ) ) {
 			$start_date = 'this_month';
 			$end_date   = false;
 		}

From 28b9ee9c8006417bed2f839e3ca777363599a138 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:47:26 -0500
Subject: [PATCH 19/83] Instantiate array before use

---
 includes/class-edd-cli.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/includes/class-edd-cli.php b/includes/class-edd-cli.php
index 4a50764..6072bf4 100644
--- a/includes/class-edd-cli.php
+++ b/includes/class-edd-cli.php
@@ -602,7 +602,8 @@ public function payments( $args, $assoc_args ) {
 					'quantity' => 1,
 					'options'  => $options
 				);
-
+				
+				$cart_details = array();
 				$cart_details[$key] = array(
 					'name'        => $download->post_title,
 					'id'          => $download->ID,

From 15bf5cc2be8d64accc43b2500fdc4c1eb05da3d7 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:55:05 -0500
Subject: [PATCH 20/83] Correct docbloc

---
 includes/admin/tracking.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/admin/tracking.php b/includes/admin/tracking.php
index b7eb6e6..168afed 100755
--- a/includes/admin/tracking.php
+++ b/includes/admin/tracking.php
@@ -197,7 +197,7 @@ public function check_for_optout( $data ) {
 	 * Get the last time a checkin was sent
 	 *
 	 * @access private
-	 * @return false/string
+	 * @return false|string
 	 */
 	private function get_last_send() {
 		return get_option( 'edd_tracking_last_send' );


From d59098be50b50442cb8b4e59fabcf1bd7b02e93e Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:58:22 -0500
Subject: [PATCH 24/83] um. What?

---
 includes/admin/discounts/class-discount-codes-table.php | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/includes/admin/discounts/class-discount-codes-table.php b/includes/admin/discounts/class-discount-codes-table.php
index 7cd3ea3..0691fc7 100755
--- a/includes/admin/discounts/class-discount-codes-table.php
+++ b/includes/admin/discounts/class-discount-codes-table.php
@@ -180,10 +180,7 @@ public function get_sortable_columns() {
 	 * @return string Column Name
 	 */
 	function column_default( $item, $column_name ) {
-		switch( $column_name ){
-			default:
-				return $item[ $column_name ];
-		}
+		return $item[ $column_name ];
 	}
 
 	/**

From f438030a65331164d0984cd55c459a382bd76e5b Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 13:59:36 -0500
Subject: [PATCH 25/83] Unused parameter

---
 includes/admin/downloads/dashboard-columns.php | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/includes/admin/downloads/dashboard-columns.php b/includes/admin/downloads/dashboard-columns.php
index 9a2477b..ec35e62 100755
--- a/includes/admin/downloads/dashboard-columns.php
+++ b/includes/admin/downloads/dashboard-columns.php
@@ -18,11 +18,10 @@
  * Defines the custom columns and their order
  *
  * @since 1.0
- * @param array $download_columns Array of download columns
  * @return array $download_columns Updated array of download columns for Downloads
  *  Post Type List Table
  */
-function edd_download_columns( $download_columns ) {
+function edd_download_columns( ) {
 	$download_columns = array(
 		'cb'                => '<input type="checkbox"/>',
 		'title'             => __( 'Name', 'edd' ),

From 1cb27fc7b16e47f727f679da135d0794cfbdfbfd Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:02:00 -0500
Subject: [PATCH 26/83] Unused parameter

---
 includes/admin/payments/payments-history.php | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/includes/admin/payments/payments-history.php b/includes/admin/payments/payments-history.php
index 5085e7d..2d37d4a 100755
--- a/includes/admin/payments/payments-history.php
+++ b/includes/admin/payments/payments-history.php
@@ -74,10 +74,9 @@ function edd_payment_history_mobile_link() {
  * @since 1.6
  *
  * @param $admin_title
- * @param $title
  * @return string
  */
-function edd_view_order_details_title( $admin_title, $title ) {
+function edd_view_order_details_title( $admin_title ) {
 	if ( 'download_page_edd-payment-history' != get_current_screen()->base )
 		return $admin_title;
 

From 5cf6607513325b237d3035fff156231ae9a1f47a Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:02:33 -0500
Subject: [PATCH 27/83] unused param

---
 includes/admin/payments/payments-history.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/admin/payments/payments-history.php b/includes/admin/payments/payments-history.php
index 2d37d4a..57caba4 100755
--- a/includes/admin/payments/payments-history.php
+++ b/includes/admin/payments/payments-history.php
@@ -110,7 +110,7 @@ function edd_view_order_details_title( $admin_title ) {
  * @param $context
  * @return string
  */
-function edd_override_edit_post_for_payment_link( $url, $post_id = 0, $context ) {
+function edd_override_edit_post_for_payment_link( $url, $post_id = 0 ) {
 
 	$post = get_post( $post_id );
 	if( ! $post )

From 7e13c0afc20f620347220e3b005f9c2942195b47 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:03:13 -0500
Subject: [PATCH 28/83] Simplify function

---
 includes/admin/reporting/class-api-requests-logs-list-table.php | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/includes/admin/reporting/class-api-requests-logs-list-table.php b/includes/admin/reporting/class-api-requests-logs-list-table.php
index 9fce0cc..97b7e89 100755
--- a/includes/admin/reporting/class-api-requests-logs-list-table.php
+++ b/includes/admin/reporting/class-api-requests-logs-list-table.php
@@ -107,10 +107,7 @@ public function get_columns() {
 	 * @return string Column Name
 	 */
 	public function column_default( $item, $column_name ) {
-		switch( $column_name ){
-			default:
-				return $item[ $column_name ];
-		}
+		return $item[ $column_name ];
 	}
 
 	/**

From a4ea2613faa72d3ebbdb176682d2218add3a6034 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:04:21 -0500
Subject: [PATCH 29/83] Unused param

---
 includes/admin/reporting/class-file-downloads-logs-list-table.php | 1 -
 1 file changed, 1 deletion(-)

diff --git a/includes/admin/reporting/class-file-downloads-logs-list-table.php b/includes/admin/reporting/class-file-downloads-logs-list-table.php
index dbc7866..422a414 100755
--- a/includes/admin/reporting/class-file-downloads-logs-list-table.php
+++ b/includes/admin/reporting/class-file-downloads-logs-list-table.php
@@ -429,7 +429,6 @@ function prepare_items() {
 		$hidden                = array(); // No hidden columns
 		$sortable              = $this->get_sortable_columns();
 		$this->_column_headers = array( $columns, $hidden, $sortable );
-		$current_page          = $this->get_pagenum();
 		$this->items           = $this->get_logs();
 		$total_items           = $edd_logs->get_log_count( $this->get_filtered_download(), 'file_download', $this->get_meta_query() );
 		$this->set_pagination_args( array(

From 313e396f6c3105ef039ed2b8e72242d83c930895 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:04:43 -0500
Subject: [PATCH 30/83] Unused variable

---
 includes/admin/reporting/class-gateway-error-logs-list-table.php | 1 -
 1 file changed, 1 deletion(-)

diff --git a/includes/admin/reporting/class-gateway-error-logs-list-table.php b/includes/admin/reporting/class-gateway-error-logs-list-table.php
index fbde478..247d157 100755
--- a/includes/admin/reporting/class-gateway-error-logs-list-table.php
+++ b/includes/admin/reporting/class-gateway-error-logs-list-table.php
@@ -207,7 +207,6 @@ public function prepare_items() {
 		$hidden                = array(); // No hidden columns
 		$sortable              = $this->get_sortable_columns();
 		$this->_column_headers = array( $columns, $hidden, $sortable );
-		$current_page          = $this->get_pagenum();
 		$this->items           = $this->get_logs();
 		$total_items           = $edd_logs->get_log_count( 0, 'gateway_error' );
 

From 1867afe34913c98a618f42dfefba33a729429c1d Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:05:11 -0500
Subject: [PATCH 31/83] Simplify function

---
 includes/admin/reporting/class-gateways-reports-table.php | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/includes/admin/reporting/class-gateways-reports-table.php b/includes/admin/reporting/class-gateways-reports-table.php
index 46cfd45..7dbb6f6 100755
--- a/includes/admin/reporting/class-gateways-reports-table.php
+++ b/includes/admin/reporting/class-gateways-reports-table.php
@@ -63,10 +63,7 @@ public function __construct() {
 	 * @return string Column Name
 	 */
 	public function column_default( $item, $column_name ) {
-		switch( $column_name ){
-			default:
-				return $item[ $column_name ];
-		}
+		return $item[ $column_name ];
 	}
 
 	/**

From acfa8ad4b665ca563b8c3b5869517f22ba8bbe4a Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:05:59 -0500
Subject: [PATCH 32/83] Unused variables

---
 includes/admin/reporting/class-sales-logs-list-table.php | 2 --
 1 file changed, 2 deletions(-)

diff --git a/includes/admin/reporting/class-sales-logs-list-table.php b/includes/admin/reporting/class-sales-logs-list-table.php
index a38b5d0..c7d0229 100755
--- a/includes/admin/reporting/class-sales-logs-list-table.php
+++ b/includes/admin/reporting/class-sales-logs-list-table.php
@@ -277,7 +277,6 @@ public function get_logs() {
 		$logs_data = array();
 		$paged     = $this->get_paged();
 		$download  = empty( $_GET['s'] ) ? $this->get_filtered_download() : null;
-		$user      = $this-> get_filtered_user();
 
 		$log_query = array(
 			'post_parent' => $download,
@@ -339,7 +338,6 @@ public function prepare_items() {
 		$hidden                = array();
 		$sortable              = $this->get_sortable_columns();
 		$this->_column_headers = array( $columns, $hidden, $sortable );
-		$current_page          = $this->get_pagenum();
 		$this->items           = $this->get_logs();
 		$total_items           = $edd_logs->get_log_count( $this->get_filtered_download(), 'sale', $this->get_meta_query() );
 

From 54910f5255cf33cd71e769b8911d17857fcb9f87 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:07:21 -0500
Subject: [PATCH 33/83] Unused variables

---
 includes/admin/reporting/pdf-reports.php | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/includes/admin/reporting/pdf-reports.php b/includes/admin/reporting/pdf-reports.php
index 59cbeff..c7919f3 100755
--- a/includes/admin/reporting/pdf-reports.php
+++ b/includes/admin/reporting/pdf-reports.php
@@ -111,7 +111,6 @@ function edd_generate_pdf( $data ) {
 			$tags = $tags ? strip_tags( $tags ) : '';
 
 			$sales = edd_get_download_sales_stats( $download->ID );
-			$link = get_permalink( $download->ID );
 			$earnings = html_entity_decode ( edd_currency_filter( edd_get_download_earnings_stats( $download->ID ) ) );
 
 			if( function_exists( 'iconv' ) ) {
@@ -186,9 +185,7 @@ function edd_draw_chart_image() {
 		$i++;
 	}
 
-	$min_earnings = 0;
 	$max_earnings = max( $earnings_array );
-	$earnings_scale = round( $max_earnings, -1 );
 
 	$data = new GoogleChartData( array(
 		$earnings_array[0],

From 980377bff24d9859212c99958a3bd77205fe27f5 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:09:59 -0500
Subject: [PATCH 34/83] unused variables

---
 includes/admin/upgrades/upgrade-functions.php | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/includes/admin/upgrades/upgrade-functions.php b/includes/admin/upgrades/upgrade-functions.php
index 6e62063..b1dc8df 100755
--- a/includes/admin/upgrades/upgrade-functions.php
+++ b/includes/admin/upgrades/upgrade-functions.php
@@ -208,12 +208,10 @@ function edd_v131_upgrades() {
 	);
 
 	$query = new WP_Query( $args );
-	$count = $query->post_count;
 	$downloads = $query->get_posts();
 
 	if ( $downloads ) {
 		$edd_log = new EDD_Logging();
-		$i = 0;
 		foreach ( $downloads as $download ) {
 			// Convert sale logs
 			$sale_logs = edd_get_download_sales_log( $download->ID, false );
@@ -314,7 +312,6 @@ function edd_v14_upgrades() {
 
 	if ( $discounts ) {
 		foreach ( $discounts as $discount_key => $discount ) {
-			$status = isset( $discount['status'] ) ? $discount['status'] : 'inactive';
 
 			$discount_id = wp_insert_post( array(
 				'post_type'   => 'edd_discount',

From ba5d502e048fd69c5d75b187a6748b8757512415 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:12:14 -0500
Subject: [PATCH 35/83] doc bloc

---
 includes/cart/functions.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/cart/functions.php b/includes/cart/functions.php
index fd7c024..2bc4886 100755
--- a/includes/cart/functions.php
+++ b/includes/cart/functions.php
@@ -473,7 +473,7 @@ function edd_get_cart_item_final_price( $item_key = 0 ) {
  * Get cart item tax
  *
  * @since 1.9
- * @param array $item Download ID
+ * @param array $download_id Download ID
  * @param array $options Cart item options
  * @param float $subtotal Cart item subtotal
  * @return float Tax amount

From e7a4932a8182bb69217032523c66a57d5f080ac4 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:13:33 -0500
Subject: [PATCH 36/83] unused params

---
 includes/cart/functions.php | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/includes/cart/functions.php b/includes/cart/functions.php
index 2bc4886..b83aaa5 100755
--- a/includes/cart/functions.php
+++ b/includes/cart/functions.php
@@ -839,11 +839,9 @@ function edd_add_collection_to_cart( $taxonomy, $terms ) {
  * @since 1.0
  * @global $post
  * @param int $cart_key Cart item key
- * @param object $post Download (post) object
- * @param bool $ajax AJAX?
  * @return string $remove_url URL to remove the cart item
  */
-function edd_remove_item_url( $cart_key, $post, $ajax = false ) {
+function edd_remove_item_url( $cart_key ) {
 
 	global $wp_query;
 

From 7827ab1975c2c67e5a84d8d9a655a580a804e542 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:14:11 -0500
Subject: [PATCH 37/83] deadcode

---
 includes/cart/functions.php | 2 --
 1 file changed, 2 deletions(-)

diff --git a/includes/cart/functions.php b/includes/cart/functions.php
index b83aaa5..dc4e945 100755
--- a/includes/cart/functions.php
+++ b/includes/cart/functions.php
@@ -982,8 +982,6 @@ function edd_is_cart_saved() {
 		return true;
 
 	}
-
-	return false;
 }
 
 /**

From ad7fa3693ac8e7443344cb2adf5fc74aff01738e Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:18:09 -0500
Subject: [PATCH 39/83] There's no param for edd_get_cart_subtotal

---
 includes/deprecated-functions.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/deprecated-functions.php b/includes/deprecated-functions.php
index fc6ff3d..9b1e743 100755
--- a/includes/deprecated-functions.php
+++ b/includes/deprecated-functions.php
@@ -257,7 +257,7 @@ function edd_get_cart_amount( $add_taxes = true, $local_override = false ) {
 
 	_edd_deprecated_function( __FUNCTION__, '1.9', 'edd_get_cart_subtotal() or edd_get_cart_total()', $backtrace );
 
-	$amount = edd_get_cart_subtotal( false );
+	$amount = edd_get_cart_subtotal( );
 	if ( ! empty( $_POST['edd-discount'] ) || edd_get_cart_discounts() !== false ) {
 		// Retrieve the discount stored in cookies
 		$discounts = edd_get_cart_discounts();

From de004c50c477276c430d41c04247bafa23ae3b49 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:19:09 -0500
Subject: [PATCH 40/83] There's no third param on edd_get_cart_item_price

---
 includes/discount-functions.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/discount-functions.php b/includes/discount-functions.php
index 953b386..ca47e42 100755
--- a/includes/discount-functions.php
+++ b/includes/discount-functions.php
@@ -1119,7 +1119,7 @@ function edd_get_cart_item_discount_amount( $item = array() ) {
 						$cart_items        = edd_get_cart_contents();
 						foreach( $cart_items as $cart_item ) {
 							if( ! in_array( $cart_item['id'], $excluded_products ) ) {
-								$item_price      = edd_get_cart_item_price( $cart_item['id'], $cart_item['options'], edd_prices_include_tax() );
+								$item_price      = edd_get_cart_item_price( $cart_item['id'], $cart_item['options'] );
 								$items_subtotal += $item_price * $cart_item['quantity'];
 							}
 						}

From e211f9265307b16812479e6f9cac182f595ff4f1 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:19:44 -0500
Subject: [PATCH 41/83] Another one with too many params for
 edd_get_cart_item_price

---
 includes/discount-functions.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/discount-functions.php b/includes/discount-functions.php
index ca47e42..156268e 100755
--- a/includes/discount-functions.php
+++ b/includes/discount-functions.php
@@ -1070,7 +1070,7 @@ function edd_get_cart_item_discount_amount( $item = array() ) {
 	global $edd_is_last_cart_item, $edd_flat_discount_total;
 
 	$amount           = 0;
-	$price            = edd_get_cart_item_price( $item['id'], $item['options'], edd_prices_include_tax() );
+	$price            = edd_get_cart_item_price( $item['id'], $item['options'] );
 	$discounted_price = $price;
 
 	// Retrieve all discounts applied to the cart

From c043e74e667e20347cfd81f7aaaaa428931ef6bc Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:22:02 -0500
Subject: [PATCH 42/83] $version isn't declared in Updater. Did_check isn't
 used or declared.

---
 includes/EDD_SL_Plugin_Updater.php | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/includes/EDD_SL_Plugin_Updater.php b/includes/EDD_SL_Plugin_Updater.php
index ab392b3..dac06fa 100644
--- a/includes/EDD_SL_Plugin_Updater.php
+++ b/includes/EDD_SL_Plugin_Updater.php
@@ -17,6 +17,7 @@ class EDD_SL_Plugin_Updater {
     private $api_data  = array();
     private $name      = '';
     private $slug      = '';
+	private $version      = '';
 
     /**
      * Class constructor.
@@ -86,8 +87,6 @@ function check_update( $_transient_data ) {
 
             if ( false !== $version_info && is_object( $version_info ) && isset( $version_info->new_version ) ) {
 
-                $this->did_check = true;
-
                 if( version_compare( $this->version, $version_info->new_version, '<' ) ) {
 
                     $_transient_data->response[ $this->name ] = $version_info;

From 83a9e08fc40e176d5dc7253b1bca6c4c12310f1f Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:29:50 -0500
Subject: [PATCH 43/83] Remove unused params/methods

---
 .scrutinizer.yml | 2 --
 1 file changed, 2 deletions(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index 7e4951f..807044a 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -43,8 +43,6 @@ checks:
         use_statement_alias_conflict: true
         unused_variables: true
         unused_properties: true
-        unused_parameters: true
-        unused_methods: true
         unreachable_code: true
         sql_injection_vulnerabilities: true
         security_vulnerabilities: true

From ae22d23b13bfca0924afe88bd6248ff1b9f00000 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:30:07 -0500
Subject: [PATCH 44/83] add some more checks

---
 .scrutinizer.yml | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index 807044a..7ea5156 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -56,4 +56,8 @@ checks:
         closure_use_modifiable: true
         avoid_useless_overridden_methods: true
         avoid_conflicting_incrementers: true
-        assignment_of_null_return: true
\ No newline at end of file
+        assignment_of_null_return: true
+        unused_methods: true
+        too_many_arguments: true
+        no_debug_code: true
+        missing_arguments: true
\ No newline at end of file

From b10e591ac49c3c47281287faa183053b58c60ec3 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:31:34 -0500
Subject: [PATCH 45/83] indentation

---
 includes/EDD_SL_Plugin_Updater.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/EDD_SL_Plugin_Updater.php b/includes/EDD_SL_Plugin_Updater.php
index dac06fa..4adaf94 100644
--- a/includes/EDD_SL_Plugin_Updater.php
+++ b/includes/EDD_SL_Plugin_Updater.php
@@ -17,7 +17,7 @@ class EDD_SL_Plugin_Updater {
     private $api_data  = array();
     private $name      = '';
     private $slug      = '';
-	private $version      = '';
+    private $version      = '';
 
     /**
      * Class constructor.

From 916c4656f3431feddac3d8af5d989a679021d5ee Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:32:27 -0500
Subject: [PATCH 46/83] Revert "Unused parameter"

This reverts commit f438030a65331164d0984cd55c459a382bd76e5b.
---
 includes/admin/downloads/dashboard-columns.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/includes/admin/downloads/dashboard-columns.php b/includes/admin/downloads/dashboard-columns.php
index ec35e62..9a2477b 100755
--- a/includes/admin/downloads/dashboard-columns.php
+++ b/includes/admin/downloads/dashboard-columns.php
@@ -18,10 +18,11 @@
  * Defines the custom columns and their order
  *
  * @since 1.0
+ * @param array $download_columns Array of download columns
  * @return array $download_columns Updated array of download columns for Downloads
  *  Post Type List Table
  */
-function edd_download_columns( ) {
+function edd_download_columns( $download_columns ) {
 	$download_columns = array(
 		'cb'                => '<input type="checkbox"/>',
 		'title'             => __( 'Name', 'edd' ),

From d75175d2d0f3c1277d900993ea3432f5bed41d0e Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:32:59 -0500
Subject: [PATCH 47/83] Revert "Unused parameter"

This reverts commit 1cb27fc7b16e47f727f679da135d0794cfbdfbfd.
---
 includes/admin/payments/payments-history.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/includes/admin/payments/payments-history.php b/includes/admin/payments/payments-history.php
index 57caba4..4d0add0 100755
--- a/includes/admin/payments/payments-history.php
+++ b/includes/admin/payments/payments-history.php
@@ -74,9 +74,10 @@ function edd_payment_history_mobile_link() {
  * @since 1.6
  *
  * @param $admin_title
+ * @param $title
  * @return string
  */
-function edd_view_order_details_title( $admin_title ) {
+function edd_view_order_details_title( $admin_title, $title ) {
 	if ( 'download_page_edd-payment-history' != get_current_screen()->base )
 		return $admin_title;
 

From 5c8c6561bf3f368a82fbe72967609aa1dde0364d Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:33:19 -0500
Subject: [PATCH 48/83] Revert "unused param"

This reverts commit 5cf6607513325b237d3035fff156231ae9a1f47a.
---
 includes/admin/payments/payments-history.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/admin/payments/payments-history.php b/includes/admin/payments/payments-history.php
index 4d0add0..5085e7d 100755
--- a/includes/admin/payments/payments-history.php
+++ b/includes/admin/payments/payments-history.php
@@ -111,7 +111,7 @@ function edd_view_order_details_title( $admin_title, $title ) {
  * @param $context
  * @return string
  */
-function edd_override_edit_post_for_payment_link( $url, $post_id = 0 ) {
+function edd_override_edit_post_for_payment_link( $url, $post_id = 0, $context ) {
 
 	$post = get_post( $post_id );
 	if( ! $post )

From ef3f4c7ad010cfc3b3c4af5553f479bf12916d71 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:34:09 -0500
Subject: [PATCH 49/83] Revert "unused params"

This reverts commit e7a4932a8182bb69217032523c66a57d5f080ac4.
---
 includes/cart/functions.php | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/includes/cart/functions.php b/includes/cart/functions.php
index dc4e945..9aceb7d 100755
--- a/includes/cart/functions.php
+++ b/includes/cart/functions.php
@@ -839,9 +839,11 @@ function edd_add_collection_to_cart( $taxonomy, $terms ) {
  * @since 1.0
  * @global $post
  * @param int $cart_key Cart item key
+ * @param object $post Download (post) object
+ * @param bool $ajax AJAX?
  * @return string $remove_url URL to remove the cart item
  */
-function edd_remove_item_url( $cart_key ) {
+function edd_remove_item_url( $cart_key, $post, $ajax = false ) {
 
 	global $wp_query;
 

From be8936ee6798ab2f9b0fd76baf4945729bf5834d Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 14:35:26 -0500
Subject: [PATCH 50/83] Removed unused parameters

---
 includes/cart/functions.php | 4 +---
 includes/cart/template.php  | 2 +-
 templates/checkout_cart.php | 2 +-
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/includes/cart/functions.php b/includes/cart/functions.php
index 9aceb7d..dc4e945 100755
--- a/includes/cart/functions.php
+++ b/includes/cart/functions.php
@@ -839,11 +839,9 @@ function edd_add_collection_to_cart( $taxonomy, $terms ) {
  * @since 1.0
  * @global $post
  * @param int $cart_key Cart item key
- * @param object $post Download (post) object
- * @param bool $ajax AJAX?
  * @return string $remove_url URL to remove the cart item
  */
-function edd_remove_item_url( $cart_key, $post, $ajax = false ) {
+function edd_remove_item_url( $cart_key ) {
 
 	global $wp_query;
 
diff --git a/includes/cart/template.php b/includes/cart/template.php
index edd14e3..6d957a8 100755
--- a/includes/cart/template.php
+++ b/includes/cart/template.php
@@ -76,7 +76,7 @@ function edd_get_cart_item_template( $cart_key, $item, $ajax = false ) {
 
 	$id = is_array( $item ) ? $item['id'] : $item;
 
-	$remove_url = edd_remove_item_url( $cart_key, $post, $ajax );
+	$remove_url = edd_remove_item_url( $cart_key );
 	$title      = get_the_title( $id );
 	$options    = !empty( $item['options'] ) ? $item['options'] : array();
 	$price      = edd_get_cart_item_price( $id, $options );
diff --git a/templates/checkout_cart.php b/templates/checkout_cart.php
index d4d80d7..7744212 100755
--- a/templates/checkout_cart.php
+++ b/templates/checkout_cart.php
@@ -42,7 +42,7 @@
 							<input type="hidden" name="edd-cart-downloads[]" value="<?php echo $item['id']; ?>"/>
 							<input type="hidden" name="edd-cart-download-<?php echo $key; ?>-options" value="<?php echo esc_attr( serialize( $item['options'] ) ); ?>"/>
 						<?php endif; ?>
-						<a class="edd_cart_remove_item_btn" href="<?php echo esc_url( edd_remove_item_url( $key, $post ) ); ?>"><?php _e( 'Remove', 'edd' ); ?></a>
+						<a class="edd_cart_remove_item_btn" href="<?php echo esc_url( edd_remove_item_url( $key ) ); ?>"><?php _e( 'Remove', 'edd' ); ?></a>
 					</td>
 					<?php do_action( 'edd_checkout_table_body_last', $item ); ?>
 				</tr>

From 9d2ccdeb948529c7b0de55e2029e58a48a3b3f65 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 15:02:49 -0500
Subject: [PATCH 52/83] docblocs

---
 includes/emails/class-edd-emails.php | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/includes/emails/class-edd-emails.php b/includes/emails/class-edd-emails.php
index 5f693dc..ab5e239 100644
--- a/includes/emails/class-edd-emails.php
+++ b/includes/emails/class-edd-emails.php
@@ -169,6 +169,8 @@ public function get_templates() {
 	 * Get the enabled email template
 	 *
 	 * @since 2.1
+	 *
+	 * @return string|null
 	 */
 	public function get_template() {
 		if ( ! $this->template ) {
@@ -205,6 +207,9 @@ public function parse_tags( $content ) {
 	 * Build the final email
 	 *
 	 * @since 2.1
+	 * @param string $message
+	 *
+	 * @return string
 	 */
 	public function build_email( $message ) {
 

From 5fd8869f875c15b38e90a9b97571f59c4bc8991f Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 15:13:28 -0500
Subject: [PATCH 53/83] Correct SL docbloc

---
 includes/EDD_SL_Plugin_Updater.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/EDD_SL_Plugin_Updater.php b/includes/EDD_SL_Plugin_Updater.php
index 4adaf94..46b20cc 100644
--- a/includes/EDD_SL_Plugin_Updater.php
+++ b/includes/EDD_SL_Plugin_Updater.php
@@ -263,7 +263,7 @@ function http_request_args( $args, $url ) {
      *
      * @param string  $_action The requested action.
      * @param array   $_data   Parameters for the API action.
-     * @return false||object
+     * @return false|object
      */
     private function api_request( $_action, $_data ) {
 

From 45c343615ffdd4a785f878dc329b01139d44d33f Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 15:21:01 -0500
Subject: [PATCH 54/83] Turn off unused_variables

---
 .scrutinizer.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index 7ea5156..9a5ed27 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -41,7 +41,7 @@ checks:
         variable_existence: true
         useless_calls: true
         use_statement_alias_conflict: true
-        unused_variables: true
+        unused_variables: false
         unused_properties: true
         unreachable_code: true
         sql_injection_vulnerabilities: true

From 9a46bb2be2b0530b3e533c43d1a49d544e93ddd5 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 15:31:11 -0500
Subject: [PATCH 55/83] fix exclude

---
 .travis.yml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/.travis.yml b/.travis.yml
index 2e0cd00..7a58825 100755
--- a/.travis.yml
+++ b/.travis.yml
@@ -21,9 +21,9 @@ matrix:
       env: WP_VERSION=latest WP_MULTISITE=1
   exclude:
     - php: hhvm
-      env: WP_VERSION=3.9.2 WP_MULTISITE=1
+      env: WP_VERSION=3.9.2 WP_MULTISITE=0
     - php: hhvm
-      env: WP_VERSION=4.0 WP_MULTISITE=1
+      env: WP_VERSION=4.0 WP_MULTISITE=0
   fast_finish: true
 
 before_script:

From 26616624e1efa1f4a79a52cdd18908ec196f0bab Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 15:43:21 -0500
Subject: [PATCH 56/83] Turn argument_type_checks off

---
 .scrutinizer.yml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index 9a5ed27..ff761b4 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -60,4 +60,5 @@ checks:
         unused_methods: true
         too_many_arguments: true
         no_debug_code: true
-        missing_arguments: true
\ No newline at end of file
+        missing_arguments: true
+        argument_type_checks: false
\ No newline at end of file

From a9e1eef9f2d5f51f9116cd2fe137f01e520118dd Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 16:11:15 -0500
Subject: [PATCH 57/83] Turn of unused params and error supression

---
 .scrutinizer.yml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index ff761b4..79fbc27 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -61,4 +61,6 @@ checks:
         too_many_arguments: true
         no_debug_code: true
         missing_arguments: true
-        argument_type_checks: false
\ No newline at end of file
+        argument_type_checks: false
+        unused_parameters: false
+        no_error_suppression: false
\ No newline at end of file

From 1277ceb68a0fde3fa81a0530ac831d58ab5e788e Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 16:13:55 -0500
Subject: [PATCH 58/83] return_doc_comments

---
 .scrutinizer.yml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index 79fbc27..0319885 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -63,4 +63,5 @@ checks:
         missing_arguments: true
         argument_type_checks: false
         unused_parameters: false
-        no_error_suppression: false
\ No newline at end of file
+        no_error_suppression: false
+        return_doc_comments: false
\ No newline at end of file

From 02f89a004658d8c023ce61170d885db316699b4b Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 16:31:39 -0500
Subject: [PATCH 59/83] more scrutinizer

---
 .scrutinizer.yml | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/.scrutinizer.yml b/.scrutinizer.yml
index 0319885..1dcd859 100644
--- a/.scrutinizer.yml
+++ b/.scrutinizer.yml
@@ -43,6 +43,7 @@ checks:
         use_statement_alias_conflict: true
         unused_variables: false
         unused_properties: true
+        unused_parameters: false
         unreachable_code: true
         sql_injection_vulnerabilities: true
         security_vulnerabilities: true
@@ -62,6 +63,6 @@ checks:
         no_debug_code: true
         missing_arguments: true
         argument_type_checks: false
-        unused_parameters: false
         no_error_suppression: false
-        return_doc_comments: false
\ No newline at end of file
+        return_doc_comments: false
+        property_assignments: false
\ No newline at end of file

From d091d404d7d46194701f161d30284b7ca761fada Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 17:04:05 -0500
Subject: [PATCH 60/83] fix docbloc

---
 includes/payments/functions.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/payments/functions.php b/includes/payments/functions.php
index 02e9e67..9174c50 100755
--- a/includes/payments/functions.php
+++ b/includes/payments/functions.php
@@ -1477,7 +1477,7 @@ function edd_delete_payment_note( $comment_id = 0, $payment_id = 0 ) {
  * Gets the payment note HTML
  *
  * @since 1.9
- * @param object/int $note The comment object or ID
+ * @param object|int $note The comment object or ID
  * @param int $payment_id The payment ID the note is connected to
  * @return string
  */

From 7341ecd760e6fde7fb0a8d4ccc9a5d53f1820721 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 17:07:04 -0500
Subject: [PATCH 61/83] create array initialization for $error in api class

---
 includes/class-edd-api.php | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/includes/class-edd-api.php b/includes/class-edd-api.php
index 0cbddeb..310c5c6 100755
--- a/includes/class-edd-api.php
+++ b/includes/class-edd-api.php
@@ -251,6 +251,7 @@ public function get_user( $key = '' ) {
 	 * @since 1.5
 	 */
 	private function missing_auth() {
+		$error = array();
 		$error['error'] = __( 'You must specify both a token and API key!', 'edd' );
 
 		$this->data = $error;
@@ -267,6 +268,7 @@ private function missing_auth() {
 	 * @return void
 	 */
 	private function invalid_auth() {
+		$error = array();
 		$error['error'] = __( 'Your request could not be authenticated!', 'edd' );
 
 		$this->data = $error;
@@ -284,6 +286,7 @@ private function invalid_auth() {
 	 * @return void
 	 */
 	private function invalid_key() {
+		$error = array();
 		$error['error'] = __( 'Invalid API key!', 'edd' );
 
 		$this->data = $error;
@@ -400,7 +403,7 @@ public function get_query_mode() {
 		) );
 
 		$query = isset( $wp_query->query_vars['edd-api'] ) ? $wp_query->query_vars['edd-api'] : null;
-
+		$error = array();
 		// Make sure our query is valid
 		if ( ! in_array( $query, $accepted ) ) {
 			$error['error'] = __( 'Invalid query!', 'edd' );
@@ -635,7 +638,7 @@ public function get_dates( $args = array() ) {
 	public function get_customers( $customer = null ) {
 
 		$customers = array();
-
+		$error = array();
 		if( ! user_can( $this->user_id, 'view_shop_sensitive_data' ) && ! $this->override ) {
 			return $customers;
 		}

From 8f1433ab18b8ec25011fae918c955f3b23c7b5b5 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 17:12:35 -0500
Subject: [PATCH 62/83] make a more sane looking default.

---
 includes/shortcodes.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/includes/shortcodes.php b/includes/shortcodes.php
index 63b3f51..11843c6 100755
--- a/includes/shortcodes.php
+++ b/includes/shortcodes.php
@@ -474,7 +474,6 @@ function edd_downloads_query( $atts, $content = null ) {
 		case 2:
 			$column_width = '50%'; break;
 		case 3:
-		default:
 			$column_width = '33%'; break;
 		case 4:
 			$column_width = '25%'; break;
@@ -482,6 +481,8 @@ function edd_downloads_query( $atts, $content = null ) {
 			$column_width = '20%'; break;
 		case 6:
 			$column_width = '16.6%'; break;
+		default:
+			$column_width = '33%'; break;
 	endswitch;
 
 	// Allow the query to be manipulated by other plugins

From e5bb5010c8e2dca3b9a69f8c55a55c17ee99baa6 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 17:27:39 -0500
Subject: [PATCH 63/83] fix docbloc

---
 includes/cart/functions.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/cart/functions.php b/includes/cart/functions.php
index dc4e945..b47dcb0 100755
--- a/includes/cart/functions.php
+++ b/includes/cart/functions.php
@@ -417,7 +417,7 @@ function edd_cart_item_price( $item_id = 0, $options = array() ) {
  * Use edd_get_cart_item_final_price()
  *
  * @since 1.0
- * @param int   $item_id Download ID number
+ * @param int   $download_id Download ID number
  * @param array $options Optional parameters, used for defining variable prices
  * @return float|bool Price for this item
  */

From 49760f6a76234ffda2758090030529a14d7e6464 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 17:31:36 -0500
Subject: [PATCH 64/83] initialize arrays

---
 includes/class-edd-api.php | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/includes/class-edd-api.php b/includes/class-edd-api.php
index 310c5c6..c8262cd 100755
--- a/includes/class-edd-api.php
+++ b/includes/class-edd-api.php
@@ -738,6 +738,8 @@ public function get_products( $product = null ) {
 				'suppress_filters' => true,
 				'paged'            => $this->get_paged()
 			) );
+			
+			$error = array();
 
 			if ( $product_list ) {
 				$i = 0;
@@ -1181,6 +1183,7 @@ public function get_discounts( $discount = null ) {
 		if( ! user_can( $this->user_id, 'manage_shop_discounts' ) && ! $this->override ) {
 			return $discount_list;
 		}
+		$error = array();
 
 		if ( empty( $discount ) ) {
 
@@ -1600,7 +1603,7 @@ private function get_token( $user_id = 0 ) {
 	private function get_default_sales_stats() {
 
 		// Default sales return
-
+		$sales = array();
 		$sales['sales']['today']         = $this->stats->get_sales( 0, 'today' );
 		$sales['sales']['current_month'] = $this->stats->get_sales( 0, 'this_month' );
 		$sales['sales']['last_month']    = $this->stats->get_sales( 0, 'last_month' );
@@ -1619,7 +1622,7 @@ private function get_default_sales_stats() {
 	private function get_default_earnings_stats() {
 
 		// Default earnings return
-
+		$earnings = array();
 		$earnings['earnings']['today']         = $this->stats->get_earnings( 0, 'today' );
 		$earnings['earnings']['current_month'] = $this->stats->get_earnings( 0, 'this_month' );
 		$earnings['earnings']['last_month']    = $this->stats->get_earnings( 0, 'last_month' );


From e46759f415e59789b89a5fc179aa4e6cc473a213 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 17:42:45 -0500
Subject: [PATCH 67/83] Another copyright date updated

---
 includes/admin/class-edd-notices.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/admin/class-edd-notices.php b/includes/admin/class-edd-notices.php
index 7519e10..d428db0 100644
--- a/includes/admin/class-edd-notices.php
+++ b/includes/admin/class-edd-notices.php
@@ -4,7 +4,7 @@
  *
  * @package     EDD
  * @subpackage  Admin/Notices
- * @copyright   Copyright (c) 2014, Pippin Williamson
+ * @copyright   Copyright (c) 2015, Pippin Williamson
  * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
  * @since       2.3
  */

From 2e433bc3b750ea13615e6c3d9eebcd20c0c3de93 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 17:45:43 -0500
Subject: [PATCH 68/83] move it up to set it

---
 includes/class-edd-api.php | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/includes/class-edd-api.php b/includes/class-edd-api.php
index c8262cd..f54a397 100755
--- a/includes/class-edd-api.php
+++ b/includes/class-edd-api.php
@@ -728,7 +728,8 @@ public function get_customers( $customer = null ) {
 	public function get_products( $product = null ) {
 
 		$products = array();
-
+		$error = array();
+		
 		if ( $product == null ) {
 			$products['products'] = array();
 
@@ -738,8 +739,6 @@ public function get_products( $product = null ) {
 				'suppress_filters' => true,
 				'paged'            => $this->get_paged()
 			) );
-			
-			$error = array();
 
 			if ( $product_list ) {
 				$i = 0;
 

From 56f76d486ff31a153a5fc1f283a3548293b67313 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 17:54:00 -0500
Subject: [PATCH 70/83] The Download class doesn't have a post_status key. cc:
 @pippinsplugins @cklosowski

---
 includes/template-functions.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/includes/template-functions.php b/includes/template-functions.php
index 8847f70..39643b3 100755
--- a/includes/template-functions.php
+++ b/includes/template-functions.php
@@ -72,12 +72,13 @@ function edd_get_purchase_link( $args = array() ) {
 	$args = wp_parse_args( $args, $defaults );
 
 	$download = new EDD_Download( $args['download_id'] );
+	$download_post = get_post( $args['download_id'] );
 
 	if( empty( $download->ID ) ) {
 		return false;
 	}
 
-	if( 'publish' !== $download->post_status && ! current_user_can( 'edit_product', $download->ID ) ) {
+	if( 'publish' !== $download_post->post_status && ! current_user_can( 'edit_product', $download->ID ) ) {
 		return false; // Product not published or user doesn't have permission to view drafts
 	}
 

From 7a74b646b24b61fdb79544734a7c37e470282501 Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 20:56:44 -0500
Subject: [PATCH 71/83] whitespace

---
 includes/EDD_SL_Plugin_Updater.php            | 2 +-
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/includes/EDD_SL_Plugin_Updater.php b/includes/EDD_SL_Plugin_Updater.php
index 46b20cc..96e6678 100644
--- a/includes/EDD_SL_Plugin_Updater.php
+++ b/includes/EDD_SL_Plugin_Updater.php
@@ -17,7 +17,7 @@ class EDD_SL_Plugin_Updater {
     private $api_data  = array();
     private $name      = '';
     private $slug      = '';
-    private $version      = '';
+    private $version   = '';
 
     /**
      * Class constructor.


From a397904228f22c0e6e0487579feb2837063e765d Mon Sep 17 00:00:00 2001
From: Chris Christoff <hello@chriscct7.com>
Date: Sat, 14 Feb 2015 21:05:50 -0500
Subject: [PATCH 72/83] whitespace

---
 includes/template-functions.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/template-functions.php b/includes/template-functions.php
index 39643b3..ad5144d 100755
--- a/includes/template-functions.php
+++ b/includes/template-functions.php
@@ -71,7 +71,7 @@ function edd_get_purchase_link( $args = array() ) {
 
 	$args = wp_parse_args( $args, $defaults );
 
-	$download = new EDD_Download( $args['download_id'] );
+	$download      = new EDD_Download( $args['download_id'] );
 	$download_post = get_post( $args['download_id'] );
 
 	if( empty( $download->ID ) ) {

From b2b93a2001b3f87cfe86570b8a3c97b03cb604c7 Mon Sep 17 00:00:00 2001
From: Chris Christoff <chriscct7@users.noreply.github.com>
Date: Sat, 14 Feb 2015 22:03:13 -0500
Subject: [PATCH 73/83] whitespace

---
 includes/admin/reporting/class-api-requests-logs-list-table.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/includes/admin/reporting/class-api-requests-logs-list-table.php b/includes/admin/reporting/class-api-requests-logs-list-table.php
index 97b7e89..84fa69a 100755
--- a/includes/admin/reporting/class-api-requests-logs-list-table.php
+++ b/includes/admin/reporting/class-api-requests-logs-list-table.php
@@ -56,7 +56,7 @@ public function __construct() {
 	 * @since 1.5
 	 * @access public
 	 *
-	 * @param string $text Label for the search box
+	 * @param string $text     Label for the search box
 	 * @param string $input_id ID of the search box
 	 *
 	 * @return void
